### ISSUE TITLE
Faster repository deploy when updating

### ISSUE THREAD

**Issue by bobheadxi** [ubclaunchpad/inertia#173]
All 3 builds are two-staged (build then deploy) - hold off on container shut down until build completes when deploying

**Comment by bobheadxi** [comment]
Reopening due to #214

**Comment by bobheadxi** [comment]
Closed #214 as dupe - copied over description: 

---

#196 (which closed #173) added some stuff that allows a build and actual deploy to be separated. The idea is to allow a build to occur while a container is still online, minimizing downtime. 

But first it needs to be enabled (because I accidentally didn't actually set it up correctly) 😅 465c55c2a29ea01193cf57b6b37e59fc21d2c022 briefly enabled it but has since been rolled back (acd4ca0fa329e2480bb93eab10890c57e955e1d4) due to it apparently not working correctly:
```
 Successfully built 7d1997877a70
)Successfully tagged bumper_bumper:latest
Build exited with status 0
Preparing to start project...
Shutting down active containers...
Starting up project...
[ERROR 500] Error response from daemon: No such container: e979b873afbb25fcd5c254b486da66ea4f003b3134c154d10bb4cdc5a3c0143a
```

It also looks like images might need pruning beforehand (see [build 591.1](https://travis-ci.org/ubclaunchpad/inertia/jobs/381500127) for example):
```
--- FAIL: TestDockerComposeIntegration (94.90s)
	Error Trace:	build_test.go:119
	Error:		Expected nil, but got: &errors.errorString{s:"Error response from daemon: Conflict. The container name \"/docker-compose\" is already in use by container \"e9f46313787025fd35cb9192ff8cde4ee69b80f8ba7aec4a56c9816a5f5d6ef7\". You have to remove (or rename) that container to be able to reuse that name."}

--- FAIL: TestDockerBuildIntegration (111.11s)
	Error Trace:	build_test.go:215
	Error:		Expected nil, but got: &errors.errorString{s:"Error response from daemon: Conflict. The container name \"/test_dockerfile\" is already in use by container \"639209f6cce6fa10cd3780814979052a94381169764d873ad08d7c2768ca3628\". You have to remove (or rename) that container to be able to reuse that name."}	

--- FAIL: TestHerokuishBuildIntegration (137.78s)
	Error Trace:	build_test.go:292
	Error:		Expected nil, but got: &errors.errorString{s:"Error response from daemon: Conflict. The container name \"/build\" is already in use by container \"10f3fd1e8b99366bd5a59a59045582026ca56b722351429186b9454ba95dd3a6\". You have to remove (or rename) that container to be able to reuse that name."}
```

tl;dr I should test my code 😢 #173 has been reopened

**Comment by bobheadxi** [comment]
Might roll this functionality into #120

**Comment by bobheadxi** [comment]
Closing this in favour of #120
